### PR TITLE
Simplify (and fix) settings for transmission module

### PR DESCRIPTION
**Pull request by k0ral**
Why use `sed` when you can use nix ? :)

### CODE REVIEW

**Comment by bjornfor** [comment]
What does transmission say if you try to update its configuration from the webUI, now that its config is no longer writeable?

I'm actually a bit ambivalent towards this change. On one side, it is arguably more correct, but on the other, I sort of like the way the current module allows to change certain settings in the webUI.

I guess if the webUI is still useable for quick and dirty configuration changes (which are reset when the service restarts), then +1.

**Comment by k0ral** [comment]
There's this error at startup:

    Couldn't save temporary file "/nix/store/4sbdg08sfx8c4j9ybqcx270xsf0pn4w5-settings.json.tmp.GXcVD8": Read-only file system (variant.c:1227)`

I've tried changing some settings through the webUI, and they are correctly taken into account for the current session, but are reset when the service restarts, as you suggested.
This behavior complies with the configuration reproducibility principle of NixOS and I think it should be considered a feature rather than a bug.

**Comment by bjornfor** [comment]
There is the option of copying the file instead of symlinking, just for the sake of getting rid of that error message.

**Comment by bjornfor** [comment]
Ehm, copying + chmod I guess (chmod +w).

**Comment by k0ral** [comment]
Indeed, copying the file removes the error message. No need to chmod though, from what I've tested.

File: nixos/modules/services/torrent/transmission.nix [hunk 1]
```diff
@@ -9,28 +9,20 @@ let
   homeDir = "/var/lib/transmission";
   downloadDir = "${homeDir}/Downloads";
   incompleteDir = "${homeDir}/.incomplete";
+  
   settingsDir = "${homeDir}/.config/transmission-daemon";
-  settingsFile = "${settingsDir}/settings.json";
+  settingsFile = pkgs.writeText "settings.json" ''
+    {
+    ${ concatStringsSep ",\n" (mapAttrsToList (name: value: ''"${name}": ${toOption value}'' ) fullSettings ) }
```
Comment (copumpkin) [inline nixos/modules/services/torrent/transmission.nix]: Doesn't Nix have a `builtins.toJSON`?

Comment (k0ral) [inline nixos/modules/services/torrent/transmission.nix]: Indeed, it's much cleaner with `builtins.toJSON`, thank you.

**Comment by k0ral** [comment]
Sorry to push, but is there any reason this PR is still pending validation ? If something is wrong, please let me know and I'll fix accordingly.

**Comment by lethalman** [comment]
Haven't tested it but seems ok, merging. FYI you can `cp --remove-destination`.

**Pull request merged**